### PR TITLE
Make compatible with RN 49+, refactored and added onHide and onShow props

### PR DESCRIPTION
**Pull request by enahum**
* Refactored to be compatible with RN 49+
* Added a handler for didHide
* Added a flag to identify if there is a tooltip visible
* Added props for onHide and onShow
* Updated readme file

@chirag04 let me know if it is ok to merge and how can we do the update of the npm library ;)

### CODE REVIEW

File: ToolTip.ios.js [hunk 1]
```diff
@@ -1,92 +1,102 @@
...
-var ViewClass = React.createClass({
-  getDefaultProps: function() {
-    return {
-      arrowDirection: 'down'
+} from 'react-native';
+
+const ToolTipMenu = NativeModules.ToolTipMenu;
+const RCTToolTipText = requireNativeComponent('RCTToolTipText', null);
+
+export let isToolTipVisible = false;
```
Comment (chirag04) [inline ToolTip.ios.js]: i would put this in the component and expose via ref. this.refs.tooltip.isToolTipVisible.

**Comment by chirag04** [comment]
lgtm otherwise

File: ToolTip.ios.js (hunk 1 above)
Comment (enahum) [reply ToolTip.ios.js]: the idea is to use it as an import.

if you have multiple tooltip components and you want to know if there is at least one of those components showing the tooltip without looping though all of them.

does it make sense?

Comment (chirag04) [reply ToolTip.ios.js]: > if you have multiple tooltip components and you want to know if there is at least one of those components showing the tooltip without looping though all of them.

That sounds like a state that should live in your application then. It's little weird to keep this global export in a library.idk

**Comment by enahum** [comment]
@chirag04 agree, I removed the `isToolTipVisible` completely

**Comment by chirag04** [comment]
@enahum i guess update the readme also and merge.

**Comment by fungilation** [comment]
This is a much bigger refactor than [my PR](https://github.com/chirag04/react-native-tooltip/pull/33), which works for me up to and including RN 0.53. Looks like a nice cleanup, if everything still works the same!

**Comment by enahum** [comment]
@fungilation yeap everything works the same ;)

@chirag04 readme updated and merging...

**Pull request merged**